dateutil/test/test_parser.py

```diff
@@ -27,6 +30,24 @@
 except ValueError:
     PLATFORM_HAS_DASH_D = False
 
+class TestConvertyear(unittest.TestCase):
+    @given(integers(min_value=100))
+    def test_convertyear(self, n):
+        self.assertEqual(n, _parser.parserinfo().convertyear(n))
+
+    @given(integers(min_value=-50,
+                    max_value=49))
+    def test_no_specified_century(self, n):
+        p = _parser.parserinfo()
+        new_year = p._year + n
+        result = p.convertyear(new_year % 100, century_specified=False)
+        self.assertEqual(result, new_year)
+
+    @given(integers(max_value=-1))
+    def test_assert_non_negative(self, n):
+        with self.assertRaises(AssertionError):
+            _parser.parserinfo().convertyear(n)
```

Comment:
`_parser` is a private module and should not be directly tested. Only test the public interface, so `dateutil.parser.parserinfo`.